cache/cache.go

```diff
@@ -226,3 +221,29 @@ func readWatchChannel(
 	}
 	return nil
 }
+
+func (c *Cache) validateWatch(key string, opts ...clientv3.OpOption) (cancelChan clientv3.WatchChan, pred KeyPredicate) {
```

Comment:
`validateWatch` should validate watch and return error, not `(cancelChan clientv3.WatchChan, pred KeyPredicate)`

Reply:
+1
Interpret the error at the call sight and then close the channel like you do currently